src/neoxp/Node/OfflineNode.cs

```diff
@@ -406,5 +407,30 @@ IReadOnlyList<ExpressStorage> ListStorages(UInt160 scriptHash)
 
         public Task<IReadOnlyList<ExpressStorage>> ListStoragesAsync(UInt160 scriptHash)
             => MakeAsync(() => ListStorages(scriptHash));
+        
+        private const byte Prefix_Contract = 8;
+        
+        bool PersistContract(ContractState state, JArray storagePairs)
```

Comment:
The logic in this method appears twice - here and in OnlineNode. We should move the core downloaded contract persistence logic somewhere that can be shared by both